### PR TITLE
Improve workspace navbar

### PR DESCRIPTION
**Pull request by nwalters512**
The main goal of this PR was to improve the color contrast between the text and the navbar background.

Bootstrap 4:

Screenshot 2024-08-19 at 12 06 13

Bootstrap 5:

Screenshot 2024-08-19 at 12 05 59

I also made a few other tweaks, explained below.

### CODE REVIEW

File: apps/prairielearn/assets/scripts/workspaceClient.ts [hunk 1]
```diff
@@ -53,7 +53,7 @@ $(function () {
 
   function setMessage(message: string) {
     console.log('message', message);
-    messageBadge.innerHTML = message;
+    messageBadge.textContent = message;
```
Comment (nwalters512) [inline apps/prairielearn/assets/scripts/workspaceClient.ts:56]: We want to treat this as text, not HTML.

File: apps/prairielearn/src/pages/workspace/workspace.html.ts [hunk 2]
```diff
@@ -44,49 +44,39 @@ export function Workspace({
       >
         ${RebootModal({ __csrf_token })} ${ResetModal({ __csrf_token })}
 
-        <nav
-          class="navbar navbar-expand-md navbar-dark bg-info align-items-center"
-          style="height:55px"
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/workspace/workspace.html.ts:49]: Artificially limiting the navbar height could cause text clipping issues if the font size is set sufficiently large.

File: apps/prairielearn/src/pages/workspace/workspace.html.ts [hunk 3]
```diff
@@ -44,49 +44,39 @@ export function Workspace({
...
+              data-target="#workspace-nav"
+            >
+              <span class="navbar-toggler-icon"></span>
+            </button>
+
             <div class="collapse navbar-collapse" id="workspace-nav">
               <ul class="navbar-nav ml-auto">
-                <li class="d-sm-none nav-item ml-2 my-1">
-                  <span class="nav-item badge badge-light">${navTitle}</span>
-                </li>
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/workspace/workspace.html.ts:89]: I don't know why this was here, but AFAICT it isn't useful, the title is already rendered elsewhere.

File: apps/prairielearn/src/pages/workspace/workspace.html.ts [hunk 4]
```diff
@@ -44,49 +44,39 @@ export function Workspace({
...
+              </h1>
+              <span class="small" style="color: #000;">
                 <i class="fa fa-laptop-code" aria-hidden="true"></i>
                 PrairieLearn Workspace
               </span>
             </div>
 
-            <div class="d-flex flex-row ml-auto align-items-center">
-              <ul class="navbar-nav flex-row">
-                <li class="mr-2">
```
Comment (nwalters512) [inline apps/prairielearn/src/pages/workspace/workspace.html.ts:66]: Semantically this isn't a list, so there's no reason to mark it up as such.

File: apps/prairielearn/src/pages/workspace/workspace.html.ts [hunk 5]
```diff
@@ -44,49 +44,39 @@ export function Workspace({
...
-            <div class="d-flex flex-column mr-3 text-white">
-              <span>
-                <h1 class="h6 font-weight-normal mb-0">
-                  <a href="${navTitleHref}" target="_blank" class="text-white">${navTitle}</a>
-                </h1>
-              </span>
-              <span class="small">
+            <div class="d-flex flex-column mr-3">
+              <h1 class="h6 font-weight-normal mb-0">
+                <a href="${navTitleHref}" target="_blank" style="color: #000;">${navTitle}</a>
```
Comment (mylesw27) [inline apps/prairielearn/src/pages/workspace/workspace.html.ts:51]: Just curious, is using `.text-dark` insufficient here (and elsewhere using `style="color: #000"`)?

**Review by mylesw27 — APPROVED** [review]
Tested locally and everything works as expected. As an added bonus, it looks like removing the height limit improves responsiveness for narrow viewports as well.

File: apps/prairielearn/src/pages/workspace/workspace.html.ts (hunk 5 above)
Comment (nwalters512) [reply apps/prairielearn/src/pages/workspace/workspace.html.ts:51]: In this case, it is in fact insufficient. `.text-dark` is actually a darkish gray, not black, and this has insufficient contrast on the background used on this page. You can plug the colors into https://webaim.org/resources/contrastchecker/ to check this:

Screenshot 2024-08-19 at 15 52 07

Whereas pure black has sufficient contrast (at least for WCAG AA, which is what we're targeting at the moment):

Screenshot 2024-08-19 at 15 52 43

With BS5 styles, black in fact passes the WCA AAA contrast requirements too:

Screenshot 2024-08-19 at 15 53 43

Comment (mylesw27) [reply apps/prairielearn/src/pages/workspace/workspace.html.ts:51]: Good to know, thanks!

**Pull request merged**